src/overlays/actors/ovl_Obj_Wind/z_obj_wind.c

```diff
@@ -28,21 +29,139 @@ ActorInit Obj_Wind_InitVars = {
+    if (WaterBox_GetSurface1(play, &play->colCtx, this->actor.world.pos.x, this->actor.world.pos.z, &ySurface,
+                             &waterbox) &&
+        (this->actor.world.pos.y < ySurface)) {
+        this->isUnderWater = true;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wind/ObjWind_Init.s")
+void ObjWind_Destroy(Actor* thisx, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wind/ObjWind_Destroy.s")
+typedef struct ObjWindStruct {
+    /* 0x0 */ s16 unk_0;
+    /* 0x2 */ s16 unk_2;
+    /* 0x4 */ s16 unk_4;
+    /* 0x6 */ s16 unk_6;
+    /* 0x8 */ s16 unk_8;
+} ObjWindStruct; // size = 0xA
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wind/ObjWind_Update.s")
+static ObjWindStruct D_80B2448C[] = {
+    { 530, 90, 3000, 2000, 0 },   { 1120, 100, 2000, 1000, 0 }, { 620, 100, 2000, 1000, 0 },
+    { 1620, 100, 2000, 1000, 0 }, { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 600, 120, 3000, 2000, 0 },  { 500, 80, 3000, 2000, 0 },
+    { 400, 60, 2000, 2000, 0 },   { 470, 70, 2000, 1000, 0 },
+};
+
+static s32 D_80B245CC = -1;
+
+void ObjWind_Update(Actor* thisx, PlayState* play) {
+    ObjWind* this = (ObjWind*)thisx;
+    ObjWindStruct* entry = &D_80B2448C[OBJ_WIND_GET_TYPE(thisx)];
+    Player* player;
+    f32 cosYTimesX;
+    f32 cosX;
+    f32 sinYTimesX;
+    Vec3f posCopy;
+    Vec3f nearestPoint;
+    Vec3f sp54;
+    f32 dist;
+    f32 distToNearestPoint;
+
+    if (D_80B245CC != DREG(85)) {
+        if ((DREG(85) >= 0) && (DREG(85) < ARRAY_COUNT(D_80B2448C))) {
+            DREG(80) = D_80B2448C[DREG(85)].unk_0;
+            DREG(81) = D_80B2448C[DREG(85)].unk_2;
+            DREG(82) = D_80B2448C[DREG(85)].unk_4;
+            DREG(83) = D_80B2448C[DREG(85)].unk_6;
+            DREG(84) = D_80B2448C[DREG(85)].unk_8;
+            D_80B245CC = DREG(85);
+        }
+    } else {
+        D_80B2448C[DREG(85)].unk_0 = DREG(80);
+        D_80B2448C[DREG(85)].unk_2 = DREG(81);
+        D_80B2448C[DREG(85)].unk_4 = DREG(82);
+        D_80B2448C[DREG(85)].unk_6 = DREG(83);
+        D_80B2448C[DREG(85)].unk_8 = DREG(84);
+    }
+    if (OBJ_WIND_GET_SWITCH_FLAG(thisx) == 0x7F || !Flags_GetSwitch(play, OBJ_WIND_GET_SWITCH_FLAG(thisx))) {
```

Comment:
```suggestion
    }

    if ((OBJ_WIND_GET_SWITCH_FLAG(thisx) == 0x7F) || !Flags_GetSwitch(play, OBJ_WIND_GET_SWITCH_FLAG(thisx))) {
```
The space is optional but I find the spacing helps between blocks. The brackets would be good though